gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/RubricPreviewPanel.java

```diff
@@ -0,0 +1,102 @@
+
+    @SpringBean(name = "org.sakaiproject.gradebookng.business.GradebookNgBusinessService")
+    private GradebookNgBusinessService businessService;
+
+    private final ModalWindow window;
+
+    public RubricPreviewPanel(final String id, final IModel<Map<String, Object>> model, final ModalWindow window) {
+        super(id, model);
+        this.window = window;
+    }
+
+    @Override
+    public void onInitialize() {
+        super.onInitialize();
+        final Map<String, Object> modelData = (Map<String, Object>) getDefaultModelObject();
+        final Long assignmentId = (Long) modelData.get("assignmentId");
+        final WebMarkupContainer sakaiRubricPreview = new WebMarkupContainer("sakai-rubric-student");
+        Assignment assignmentNow = businessService.getAssignment(assignmentId);
+        if(assignmentNow!=null && assignmentNow.isExternallyMaintained()){  //this is an externally-maintained item from Assignments
+            sakaiRubricPreview.add(AttributeModifier.append("entity-id", extractAssignmentId(assignmentNow.getExternalId())));  //rubric association needs Assignment id, not Gradebook id
+            sakaiRubricPreview.add(AttributeModifier.append("tool-id", RubricsConstants.RBCS_TOOL_ASSIGNMENT));
+        } else {
+            sakaiRubricPreview.add(AttributeModifier.append("entity-id", assignmentId));
+            sakaiRubricPreview.add(AttributeModifier.append("tool-id", RubricsConstants.RBCS_TOOL_GRADEBOOKNG));
+        }
+        sakaiRubricPreview.add(AttributeModifier.append("token", rubricsService.generateJsonWebToken(RubricsConstants.RBCS_TOOL_GRADEBOOKNG)));
+        sakaiRubricPreview.add(AttributeModifier.append("instructor", "true"));
+        add(sakaiRubricPreview);
+        final GbAjaxButton done = new GbAjaxButton("done") {
+            @Override
+            public void onSubmit(final AjaxRequestTarget target, final Form<?> form) {
+                RubricPreviewPanel.this.window.close(target);
+            }
+        };
+        done.setDefaultFormProcessing(false);
+        add(done);
+        this.window.setInitialWidth(1100);
+        Assignment assignment = businessService.getAssignment(assignmentId);
+        RubricPreviewPanel.this.window.setTitle(this.getString("label.rubric.preview") + assignment.getName());
+    }
+
+    public void renderHead(final IHeaderResponse response) {
+        final String version = PortalUtils.getCDNQuery();
+        response.render(StringHeaderItem.forString(
+                "<script src=\"/webcomponents/rubrics/sakai-rubrics-utils.js" + version + "\"></script>"));
+        response.render(StringHeaderItem.forString(
+                "<script type=\"module\" src=\"/webcomponents/rubrics/rubric-association-requirements.js" + version + "\"></script>"));
+    }
+
+    private String extractAssignmentId(String externalId){
+        if (externalId==null || !externalId.contains("/")){ //make sure we have an ID that exists and contains slashes
```

Comment:
Technically you don't need this contains check because split will return a one-element array with the original string in it if there is no match.

Reply:
Changed this as you said and it works fine.